pkg/k8s/k8s.go

```diff
@@ -78,13 +78,19 @@ type Image struct {
 	Image []string `json:"image,omitempty" yaml:"image,omitempty"`
 }
 
-func (name *Name) Get(dataMap string) (string, error) {
+func (name *Name) Get(dataMap string, log *logrus.Logger) (string, error) {
 	var kindYaml map[string]interface{}
 	if err := yaml.Unmarshal([]byte(dataMap), &kindYaml); err != nil {
 		return "", err
 	}
 
 	if len(kindYaml) != 0 {
+		metadata, metadataExists := kindYaml["metadata"].(map[string]interface{})
+		if !metadataExists {
+			log.Warn("failed to get 'metadata' from the manifest")
+			return "", nil
+		}
+				
 		value, failedManifest := kindYaml["metadata"].(map[string]interface{})["name"].(string)
```

Comment:
since we are already capturing the typecasted `metadata` can we use the same here rather than type casting it again?

Reply:
sure